NOpenCL.Test/Intel/Optimization.cs

```diff
@@ -102,16 +102,13 @@ public void TestOptimization(int taskSize, int localWorkSize, bool warming, bool
                         Console.WriteLine("Run-time determines optimal workgroup size");
                     }
 
-                    IntPtr workGroupSizeMaximum = kernel.GetWorkGroupSize(device);
-                    Console.WriteLine("Maximum workgroup size for this kernel  {0}", workGroupSizeMaximum.ToInt64());
+                    ulong workGroupSizeMaximum = kernel.GetWorkGroupSize(device);
+                    Console.WriteLine("Maximum workgroup size for this kernel  {0}", workGroupSizeMaximum);
 
                     if (warming)
                     {
                         Console.Write("Warming up OpenCL execution...");
-                        using (commandQueue.EnqueueNDRangeKernel(kernel, new[] { (IntPtr)globalWorkSize }, autoGroupSize ? null : new[] { (IntPtr)localWorkSize }))
-                        {
-                        }
-
+                        commandQueue.EnqueueNDRangeKernel(kernel, new[] { (IntPtr)globalWorkSize }, autoGroupSize ? null : new[] { (IntPtr)localWorkSize });
```

Comment:
:exclamation: Not only can we not do this here, but people using this library must make sure and follow the same pattern. It's the downside of returning the event handle from each operation.

:bulb: I'm going to file a separate issue to reconsider this pattern, since it's ugly and likely to lead to incorrect usage scenarios.